corehq/apps/reports_core/filters.py

```diff
@@ -352,7 +352,8 @@ class DynamicChoiceListFilter(BaseFilter):
     template = 'reports_core/filters/dynamic_choice_list_filter/dynamic_choice_list.html'
     javascript_template = 'reports_core/filters/dynamic_choice_list_filter/dynamic_choice_list.js'
 
-    def __init__(self, name, field, datatype, label, show_all, url_generator, choice_provider, css_id=None):
+    def __init__(self, name, field, datatype, label, show_all, url_generator, choice_provider,
+                 ancestor_expression={}, css_id=None):
```

Comment:
You shouldn't set a [default argument to a mutable datatype](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments). Rather use
```python
    def __init__(  # ...
            ancestor_expression=None,  # ...
        # ...
        self.ancestor_expression = ancestor_expression or {}
        # ^^^ Assumes ancestor_expression must be truthy
        # *or* self.ancestor_expression = {} if ancestor_expression is None else ancestor_expression
        # ^^^ Allows values like 0 and ''

Reply:
Ha, thanks!

Reply:
Updated